### PR TITLE
Support nested focusable elements in VisuallyHidden with isFocusable option

### PR DESCRIPTION
**Pull request by snowystinger**
Closes https://github.com/adobe/react-spectrum/issues/3571

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

File: packages/@react-aria/visually-hidden/src/VisuallyHidden.tsx [hunk 1]
```diff
@@ -13,7 +13,7 @@
 import {DOMAttributes} from '@react-types/shared';
 import {mergeProps} from '@react-aria/utils';
 import React, {CSSProperties, JSXElementConstructor, ReactNode, useMemo, useState} from 'react';
-import {useFocus} from '@react-aria/interactions';
+import {useFocus, useFocusWithin} from '@react-aria/interactions';
```
Comment (LFDanLu) [inline packages/@react-aria/visually-hidden/src/VisuallyHidden.tsx:16]: lint
```suggestion
import {useFocusWithin} from '@react-aria/interactions';
```

File: packages/@react-aria/visually-hidden/src/VisuallyHidden.tsx [hunk 2]
```diff
@@ -57,9 +57,9 @@ export function useVisuallyHidden(props: VisuallyHiddenProps = {}): VisuallyHidd
   } = props;
 
   let [isFocused, setFocused] = useState(false);
-  let {focusProps} = useFocus({
+  let {focusWithinProps} = useFocusWithin({
```
Comment (LFDanLu) [inline packages/@react-aria/visually-hidden/src/VisuallyHidden.tsx:60]: This makes a lot more sense now, forgot `useFocus` ignores focus events on children. The `onFocusChange` I was seeing was from `useFocusRing` derp

**Review by LFDanLu — COMMENTED** [review]
behavior looks good to me, tested a example locally and things looks good. Just a lint issue to resolve

File: packages/@react-aria/visually-hidden/src/VisuallyHidden.tsx (hunk 2 above)
Comment (snowystinger) [reply packages/@react-aria/visually-hidden/src/VisuallyHidden.tsx:60]: which can also take a 'within' parameter

**Review by LFDanLu — APPROVED** [review]
LGTM

**Review by devongovett — APPROVED** [review]

**Pull request merged**